scripts/a11y/process_a11y.py

```diff
@@ -0,0 +1,144 @@
+                json.dump(data, f, indent=2)
+
+    def clean_row(self, row):
+        domain = row[0].lower()
+        agency = self.domain_to_agency.get(domain, 'N/A')
+        code = row[4]
+
+        return {
+            'domain': domain,
+            'agency': agency,
+            'branch': self.agency_to_branch.get(agency, 'Executive'),
+            'error': self.get_error_category(code),
+            'error_details': {
+                'code': code,
+                'typeCode': row[3],
+                'message': row[5],
+                'context': row[6],
+                'selector': row[7],
+            }
+
+        }
+
+    def make_a11y_data(self, data):
+        results = defaultdict(lambda: defaultdict(list))
+        for d in data:
+            results[d['domain']][d['error']].append(d['error_details'])
+
+        # using json de/encode to convert defaultdicts back to dicts
+        return {'data': json.loads(json.dumps(results))}
+
+    def make_agency_data(self, data):
+        # first, group domain stats by agency
+        data_by_agency = defaultdict(list)
+        for d in self.make_domain_data(data)['data']:
+            data_by_agency[d['agency']].append(d)
+
+        # then, compute summary stats across groups
+        results = []
+        for agency, domain_stats in data_by_agency.items():
+            pages = len(domain_stats)
+            total_errors = sum(d['errors'] for d in domain_stats)
+            entry = {
+                'agency': agency,
+                'pages_count': pages,
+                'Average Errors per Page': (
+                    'n/a' if pages == 0 else float(total_errors) / pages
+                )
+            }
+            # add in averages by error category
+            entry.update({
+                e: mean([d['errorlist'][e] for d in domain_stats])
+                for e in self.ERRORS.values()
+            })
+            results.append(entry)
+
+        return {'data': results}
+
+    def make_domain_data(self, data):
+        results = {}
+        for d in data:
+            dom = d['domain']
+            if dom not in results:
+                results[dom] = {
+                    'agency': d['agency'],
+                    'branch': d['branch'],
+                    'canonical': dom,
+                    'domain': dom,
+                    'errors': 0,
+                    'errorlist': {e: 0 for e in self.ERRORS.values()}
+                }
+            else:
+                results[dom]['errors'] += 1
+                results[dom]['errorlist'][d['error']] += 1
+
+        return {'data': list(results.values())}
+
+    def get_error_category(self, code):
+        error_id = code.split('.')[2].split('Guideline')[1]
+        return self.ERRORS.get(error_id, 'Other Errors')
+
+    @staticmethod
+    def read_csv(filename):
+        with open(filename, 'r') as f:
+            return [row for row in csv.reader(f)]
+
+
+if __name__ == '__main__':
+    a11y_filename = 'tests/data/a11y.csv'
```

Comment:
yeah, this should be changed to be an overrideable argument (or we can just pick a location and say they have to be there and named properly to run)